resources.go

```diff
@@ -1073,6 +1072,12 @@ func Provider() tfbridge.ProviderInfo {
 							return strings.ToLower(name)
 						},
 					}),
+					"source": {
+						Name: "content",
+						Asset: &tfbridge.AssetTranslation{
+							Kind: tfbridge.FileAsset,
+						},
```

Comment:
@mikhailshilkov 

One thing I realized after reviewing some code that currently uses `ZipBlob` is that the pair of changes here leads to there being no code that can be written which works on both 1.0 and 2.0.  That's a particualrly unfortunate style of breaking change.  

I wonder if we might be able to have a cleaner transition is we instead:
1. Kept this property named `source`
2. Had it support an `AltType` of `string`.  

I'm not sure we are able to support (2) yet today unfortunately.  

Other thoughts?  Or is the current break truly the right thing to do?

Reply:
You are right.

We could keep the property named `source` instead of `content`. This is not aligned to what we did in AWS, but doesn't seem important. Interestingly, it seems that we can assign a `string` to this property, at least in TypeScript:

``` ts
const sampleFile = new azure.storage.Blob("test.html", {
    name: "test.html",
    storageAccountName: storageAccount.name,
    storageContainerName: storageContainer.name,
    type: "Block",
    content: "./test.html",
    contentType: "text/html",
});
```

However, we probably don't want people to do that, because in this case Pulumi will not watch the content changes in the file as opposed to `FileAsset`.

Now, in `1.0` one couldn't upload a folder archive with `Blob`: you had to use `ZipBlob`. If we want, we could keep `ZipBlob` around in 2.0 (marked as deprecated) which would help with the transition of folder archives.

So, the most compatible way seems to be:
1. Rename `content` back to `source`. Basically, give up we will ever change it.
2. Restore `ZipBlob` as a deprecated resource (well, we still don't have deprecation 1st class support, do we?).

Reply:
I am okay with doing both (1) and (2) in this list. It's less progress on migrating forward, but helps ensure this is not too disruptive for now.